sql/src/main/java/io/crate/metadata/doc/DocSchemaInfo.java

```diff
@@ -126,37 +132,28 @@ public DocSchemaInfo(final String schemaName,
                          ClusterService clusterService,
                          Functions functions,
                          UserDefinedFunctionService udfService,
+                         ViewInfoFactory viewInfoFactory,
                          DocTableInfoFactory docTableInfoFactory) {
         this.functions = functions;
         this.schemaName = schemaName;
         this.clusterService = clusterService;
         this.udfService = udfService;
+        this.viewInfoFactory = viewInfoFactory;
         this.docTableInfoFactory = docTableInfoFactory;
     }
 
-    private DocTableInfo innerGetTableInfo(String tableName) {
-        return docTableInfoFactory.create(new TableIdent(schemaName, tableName), clusterService.state());
-    }
-
     @Override
     public TableInfo getTableInfo(String name) {
         try {
-            return docTableByName.computeIfAbsent(name, this::innerGetTableInfo);
+            return docTableByName.computeIfAbsent(name, n -> docTableInfoFactory.create(new TableIdent(schemaName, n), clusterService.state()));
         } catch (ResourceUnknownException e) {
             return null;
         }
     }
 
     private Collection<String> tableNames() {
         Set<String> tables = new HashSet<>();
-
-        Stream.of(clusterService.state().metaData().getConcreteAllIndices())
-            .filter(NO_BLOB)
-            .map(IndexParts::new)
-            .filter(indexParts -> !indexParts.isPartitioned())
-            .filter(indexParts -> indexParts.matchesSchema(schemaName))
-            .map(IndexParts::getTable)
-            .forEach(tables::add);
+        applyFiltersAndAdd(Stream.of(clusterService.state().metaData().getConcreteAllIndices()), tables);
```

Comment:
Hm, `applyFilters` is kinda unclear in what filters are applied, maybe `extractTablesFromIndices` would be clearer.